hs_core/views/utils.py

```diff
@@ -1050,6 +1043,63 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
     hydroshare.utils.resource_modified(resource, user, overwrite_bag=False)
 
 
+def ingest_bag(resource, bag_file, user):
+    """
+    Ingests a zipped bagit archive of a hydroshare reosource that has been uploaded to the resource
+    :param resource: The CompositeResource to ingest the bag into
+    :param bag_file: The ResourceFile of the zipped bag in the resource
+    :param user: The HydroShare user object to do the action as
+    """
+    from hs_file_types.utils import identify_metadata_files, ingest_metadata_files
+
+    istorage = resource.get_irods_storage()
+    zip_with_full_path = os.path.join(resource.file_path, bag_file.short_path)
+
+    # unzip to a temporary folder=
+    unzip_path = istorage.unzip(zip_with_full_path, unzipped_folder=uuid4().hex)
+    delete_resource_file(resource.short_id, bag_file.id, user)
+
+    # list all files to be moved into the resource
+    unzipped_files = listfiles_recursively(istorage, unzip_path)
+
+    res_files = []
+    for unzipped_file in unzipped_files:
+        res_files.append(IrodsFile(unzipped_file, istorage))
+    res_files, meta_files, map_files = identify_metadata_files(res_files)
+
+    # filter res_files to only files in the data/contents directory
+    data_contents_dir = os.path.join("data", "contents")
+    res_files = [res_file for res_file in res_files if res_file.name.count(data_contents_dir) > 1]
+
```

Comment:
Do we really need lines 1070-1072? Should ```res_files``` returned from identify_metadata_files() already only contain resource files in data/contents directory? If not, I think it needs to unless there are reasons I am not aware of.

Reply:
This just ensures files only in the data/contents directory are included (excludes bagit files).

Reply:
Yeah, I get that, and wondering whether the check needs to go to identify_metadata_files() so the returned res_files from it only includes files in the data/contentsxcluding bagit files. I am not sure whether there are use cases that bagit-related files are needed, but I cannot think of any at the moment, and if that is the case, it'd be better to move the exclusion of bagit-related files logic into the function itself so that the callers don't have to do this logic.

Reply:
The way identify_metadata_files() is written, it's only for identifying hydroshare specific metadata files, so the *metadata.xml and *map.xml files.  It currently isn't aware of the bagit metadata files.  I see what you're saying for readability, let me give it some thought before I say I will or won't do it.

Reply:
Looking at it more, we have to consider the directory because there could be files with the same name as the bagit files (readme.txt).  The HydroShare metadata files are reserved file names, so it is not a problem for them.  I can't just push the data/contents filter to identify_metadata_files because it is also used by unzip and the ingest_metadata endpoint.  Everything pushed through those routes is assumed to be in the data/contents directory of the resource (i.e. a file you unzip won't have a data/contents directory inside of it but the contents will be unzipped to the data/contents directory).  The ingest_bag function is unique in that it has the full structure of the resource that has to be considered when unzipped.